test/micro/org/openjdk/bench/jdk/incubator/vector/BlackScholes.java

```diff
@@ -0,0 +1,190 @@
+import java.util.function.IntUnaryOperator;
+import org.openjdk.jmh.infra.Blackhole;
+
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.SECONDS)
+@State(Scope.Thread)
+@Warmup(iterations = 3, time = 5)
+@Measurement(iterations = 3, time = 5)
+@Fork(value = 1, jvmArgsPrepend = {"--add-modules=jdk.incubator.vector"})
+public class BlackScholes {
+
+    @Param("1024")
+    int size;
+
+
+    float[] s0; // Stock Price
+    float[] x;  // Strike Price
+    float[] t;  // Maturity
+    float[] call;
+    float[] put;
+    float r;    // risk-neutrality
+    float sig;  // volatility
+    Random rand;
+
+
+    float randFloat(float low, float high) {
+       float val = rand.nextFloat()/Float.MAX_VALUE;
+       return (1.0f - val) * low + val * high;
+    }
+
+    float[] fillRandom(float low, float high) {
+        float[] array = new float[size];
+        for (int i = 0; i < array.length; i++) {
+            array[i] = randFloat(low, high);
+        }
+        return array;
+    }
+
+    @Setup
+    public void init() {
+        rand = new Random();
+        s0 = fillRandom(5.0f, 30.0f);
+        x  = fillRandom(1.0f, 100.0f);
+        t  = fillRandom(0.25f, 10.0f);
+        r = 0.02f;
+        sig = 0.30f;
+        call = new float[size];
+        put = new float[size];
+    }
+
+    static float Y = 0.2316419f;
+    static float A1 = 0.31938153f;
+    static float A2 = -0.356563782f;
+    static float A3 = 1.781477937f;
+    static float A4 = -1.821255978f;
+    static float A5 = 1.330274429f;
+    static float PI = (float)Math.PI;
+
+    float cdf(float inp) {
+        float x = inp;
+        if (inp < 0f) {
+            x = -inp;
+        }
+
+        float term = 1f / (1f + (Y * x));
+        float term_pow2 = term * term;
+        float term_pow3 = term_pow2 * term;
+        float term_pow4 = term_pow2 * term_pow2;
+        float term_pow5 = term_pow2 * term_pow3;
+
+        float part1 = (1f / (float)Math.sqrt(2f * PI)) * (float)Math.exp((-x * x) * 0.5f);
+
+        float part2 = (A1 * term) +
+                      (A2 * term_pow2) +
+                      (A3 * term_pow3) +
+                      (A4 * term_pow4) +
+                      (A5 * term_pow5);
+
+        if (inp >= 0f)
+            return 1f - part1 * part2;
+        else
+            return part1 * part2;
+
+    }
+
+    public void scalar_black_scholes_kernel(int off) {
+        float sig_sq_by2 = 0.5f * sig * sig;
+        for (int i = off; i < size; i++ ) {
+            float log_s0byx = (float)Math.log(s0[i] / x[i]);
+            float sig_sqrt_t = sig * (float)Math.sqrt(t[i]);
+            float exp_neg_rt = (float)Math.exp(-r * t[i]);
+            float d1 = (log_s0byx + (r + sig_sq_by2) * t[i])/(sig_sqrt_t);
+            float d2 = d1 - sig_sqrt_t;
+            call[i] = s0[i] * cdf(d1) - exp_neg_rt * x[i] * cdf(d2);
+            put[i]  = call[i] + exp_neg_rt - s0[i];
+       }
+    }
+
+    @Benchmark
+    public void scalar_black_scholes(Blackhole bh) {
```

Comment:
So there is a `Blackhole` argument here, but what is actually blackholed? AFAICS, stores to `call` and `put` are roughly okay to avoid dead-code elimination on these paths.